src/main/kotlin/SnakeGame.kt

```diff
@@ -44,11 +45,11 @@ class SnakeGame {
         if (event.type != KeyDown)
             return false
         when (event.key.keyCode) {
-            116500987904 -> return true                                        // Esc
-            163745628160 -> if (snake.yv == 0) { snake.xv = 0; snake.yv = -1 } // Up
-            168040595456 -> if (snake.xv == 0) { snake.xv = 1; snake.yv = 0 }  // Right
-            172335562752 -> if (snake.yv == 0) { snake.xv = 0; snake.yv = 1 }  // Down
-            159450660864 -> if (snake.xv == 0) { snake.xv = -1; snake.yv = 0 } // Left
+            116500987904 -> return true                                              // Esc
+            163745628160 -> if (snake.lastDirection != DOWN) { snake.direction = UP }    // Up
+            168040595456 -> if (snake.lastDirection != LEFT) { snake.direction = RIGHT } // Right
+            172335562752 -> if (snake.lastDirection != UP) { snake.direction = DOWN }    // Down
+            159450660864 -> if (snake.lastDirection != RIGHT) { snake.direction = LEFT } // Left
```

Comment:
…then here I can check for the direction last moved instead of the current direction